### PR TITLE
small visual cueplayer tweaks with state

### PR DESCRIPTION
**Pull request by joelhooks**

### CODE REVIEW

File: src/components/player/cue-bar.tsx [hunk 1]
```diff
@@ -79,8 +80,10 @@ const NoteCue: React.FC<any> = ({
   actions,
   player,
 }) => {
-  const playerPrefs = useEggheadPlayerPrefs()
-  const setVisible = useCue(cue, actions)
+  const {setPlayerPrefs} = useEggheadPlayerPrefs()
+  const [visible, setVisible] = React.useState(false)
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:84]: this state is really local so using it that way instead

File: src/components/player/cue-bar.tsx [hunk 2]
```diff
@@ -92,37 +95,40 @@ const NoteCue: React.FC<any> = ({
     setVisible(false)
   }
 
+  React.useEffect(() => {
+    setVisible(player.activeMetadataTrackCues.includes(cue) && !player.seeking)
+  }, [player.activeMetadataTrackCues, player.seeking, cue])
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:100]: monitor these props and update when they change

File: src/components/player/cue-bar.tsx [hunk 3]
```diff
@@ -92,37 +95,40 @@ const NoteCue: React.FC<any> = ({
...
+  }, [visible, setPlayerPrefs])
 
   return (
     <Tippy
       placement="top"
       theme="light"
-      maxWidth={800}
+      maxWidth={300}
       appendTo="parent"
+      offset={[0, 30]}
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:125]: move it above the seekbar

File: src/components/player/cue-bar.tsx [hunk 4]
```diff
@@ -92,37 +95,40 @@ const NoteCue: React.FC<any> = ({
...
+      interactive={true}
       content={
         <div className="p-2">
-          <div className="line-clamp-2">{note}</div>
-          {/* <ReactMarkdown className="prose prose-sm dark:prose-dark max-w-none"> */}
-          {/* {note.description}</div> */}
-          {/* {truncate(note.description, {length: 220, separator: '...'})} */}
-          {/* </ReactMarkdown> */}
+          <div className="line-clamp-6 prose-sm prose">
+            <ReactMarkdown>{note}</ReactMarkdown>
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:130]: markdown format

File: src/pages/video-test.tsx [hunk 5]
```diff
@@ -184,7 +185,12 @@ const NotesTabContent: React.FC<{cues: VTTCue[]}> = ({cues}) => {
                 </ReactMarkdown>
               )}
               {cue.startTime && (
-                <div className="w-full flex items-baseline justify-end pt-3 text-gray-900 dark:text-white">
+                <div
+                  onClick={() => {
+                    actions.seek(cue.startTime)
```
Comment (joelhooks) [inline src/pages/video-test.tsx:190]: nice to click time and jump as expected

File: src/components/EggheadPlayer/use-egghead-player.ts [hunk 6]
```diff
@@ -258,10 +258,10 @@ export const useEggheadPlayerPrefs = () => {
     setPlayerPrefs(getPlayerPrefs())
   }, [])
 
-  const setPlayerPrefsOptions = (options: any) => {
+  const setPlayerPrefsOptions = React.useCallback((options: any) => {
     console.debug('setting player prefs', {options})
     setPlayerPrefs(savePlayerPrefs(options))
-  }
+  }, [])
```
Comment (DrShpongle) [inline src/components/EggheadPlayer/use-egghead-player.ts:264]: Should we put `options` here?

**Pull request merged**